### PR TITLE
nxos_install_os: Fix nxapi local failures

### PR DESCRIPTION
**Pull request by mikewiebe**
##### SUMMARY
Fixes the following issues for `nxos_install_os` in the devel branch
    * Recent refactoring in module_utils broke upgrades using nxapi local
    * Refactored integration tests to make them more flexible for choosing connection type

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
nxos_install_os

##### ADDITIONAL INFORMATION

Please cherry-pick into stable-2.8

### CODE REVIEW

**Comment by mikewiebe** [comment]
@trishnaguha I fixed the conflicts with the base branch here.

File: lib/ansible/module_utils/network/nxos/nxos.py [hunk 1]
```diff
@@ -429,19 +429,43 @@ def _send(commands, output):
...
             device_info = self.get_device_info()
             if '9K' not in device_info.get('network_os_platform', ''):
                 self._module.fail_json(msg='replace is supported only on Nexus 9K devices')
             commands = 'config replace {0}'.format(replace)
 
         commands = to_list(commands)
-        msg = self.send_request(commands, output='config', check_status=True,
-                                return_error=return_error, opts=opts)
+        try:
+            resp, msg_timestamps = self.send_request(commands, output='config', check_status=True,
```
Comment (trishnaguha) [inline lib/ansible/module_utils/network/nxos/nxos.py]: As the return_timestamps is reverted, `msg_timestamps` can be now ignored here.

Comment (trishnaguha) [inline lib/ansible/module_utils/network/nxos/nxos.py]: As the `return_timestamps` is reverted, `msg_timestamps` is not needed anymore. lgtm otherwise.

**Review by trishnaguha — CHANGES_REQUESTED** [review]

**Comment by trishnaguha** [comment]
@mikewiebe We are merging 2.8 PRs now, will you be able to get this PR ready if this is required in 2.8.1?

**Comment by mikewiebe** [comment]
> @mikewiebe We are merging 2.8 PRs now, will you be able to get this PR ready if this is required in 2.8.1?

@trishnaguha Yes, I can do that.  Will tomorrow be too late for these changes?

**Comment by mikewiebe** [comment]
@trishnaguha Finally got around to removing the `msg_timestamps`. This should be good to go now.

**Review by Qalthos — APPROVED** [review]

**Review by trishnaguha — APPROVED** [review]

**Pull request merged**